docs/guides/latest/hooks/overview.md

```diff
@@ -0,0 +1,86 @@
+  <figcaption align="center">Fig: Hook Execution flow in sidecar model</figcaption>
+  </figure>
+
+- **Job Model:** In Job model, `httpGet`, `httpPost` and `tcpSocket` are executed by the backup/restore job. However, the `exec` hook is executed in the targeted application pod. In order to determine the targeted application pod, Stash uses the `Service` specified in the respective `AppBinding` crd. It first determines the endpoints of the Service. Then, it executes  the hook into one of the pod pointed by those endpoints. Hence, if the `AppBinding` does not have the respective `Service` name specified, it is not possible for Stash to execute the `exec` hook. The hook execution flow in job model is shown in the following diagram:
+
+  <figure align="center">
+    <img alt="Hook Execution flow in job model" src="/docs/images/guides/latest/hooks/job-model.svg">
+  <figcaption align="center">Fig: Hook Execution flow in job model</figcaption>
+  </figure>
+
+- **Batch Backup:** In batch backup using `BackupBatch` object, the global hooks are executed by the Stash operator itself. When Stash operator completes executing the global pre-task hook, the individual targets start executing their local pre-task hook. Then, they complete their backup process and executes their local post-task hook. Finally, the Stash operator executes global post-task hooks. The hook execution flow in batch backup is shown in the following diagram:
+
+<figure align="center">
+  <img alt="Hook Execution flow in batch backup" src="/docs/images/guides/latest/hooks/batch-backup.svg">
+<figcaption align="center">Fig: Hook Execution flow in batch backup</figcaption>
+</figure>
+
+## Hook's Behaviors
+
+Now, we are going to discuss what will happen when a hook fails or backup/restore process fails.
+
+- **Pre-Task Hook Failed:** If a pre-task hook fails to execute, the rest of the backup/restore process will be skipped and the respective `BackupSession`/`RestoreSession` will be marked as `Failed`. You may see following the things to happen in addition to skipping the backup process:
+  - **Backup Sidecar:** If the pre-task hook fails in the backup sidecar, the sidecar will just log the failure and continue watching for `BackupSession` for the next backup.
+  - **Restore Init-Container:** If the pre-task hook fails in restore init-container, the container will crash. Hence, your workload will be stuck in the initialization phase.
+  - **Backup or Restore Job:** If the pre-task hook fails in backup or restore job, the container will fail. Hence, the job will never go to the completed stage. You may see the job to create multiple pods to retry.
+
+- **Backup/Restore Process Failed:** The post-task hook will be executed even if the backup/restore process failed. This is to cover the scenario where you have paused your application in a pre-task hook and want to resume it in a post-task hook.
```

Comment:
`You may see the job to create multiple pods to retry.`

Should we allow retry? We may disable retry and fail the Job. I think that sounds better?

Reply:
We already set `backOffLimit` to 1. But, it doesn't work. :(
https://github.com/stashed/stash/blob/ba1733ebc9bcc86524613c75ef82bc2a6b729c73/pkg/controller/backup_session.go#L312

Reply:
https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#handling-pod-and-container-failures

`Note that even if you specify .spec.parallelism = 1 and .spec.completions = 1 and .spec.template.spec.restartPolicy = "Never", the same program may sometimes be started twice.`

I guess there is nothing we can do.